tests/test_utilities.py

```diff
@@ -325,21 +326,31 @@ def check_generate_predictions(use_feature_hashing=False,
 
     # now use Predictor to generate the predictions and make
     # sure that they are the same as before saving the model
-    p = gp.Predictor(model_file, threshold=threshold)
+    p = gp.Predictor(model_file, threshold=threshold,
+                     return_all_probabilities=all_probs)
     predictions_after_saving = p.predict(test_fs)
 
     eq_(predictions, predictions_after_saving)
 
 
 def test_generate_predictions():
+    possibilities = [
+        (True, True, True, False), (True, True, False, False),
+        (True, False, True, False), (True, False, True, True),
+        (True, False, False, False), (True, False, False, True),
+        (False, True, True, False), (False, True, False, False),
+        (False, False, True, False), (False, False, True, True),
+        (False, False, False, False), (False, False, False, True)]
```

Comment:
Why not just `product([True, False], [True, False], [True, False], [True, False])`?

Reply:
Because `threshold` and `all_probabilities` are mutually exclusive. I could do the product and then filter out the cases I don't want to test, or add a test for those cases to verify that an error is raised, I suppose.

Reply:
Yeah, the latter would a better option IMO.